feat: change output if all diff output is suppressed

TheMeier opened the pull request:
If a diff exists but is suppressed in the output change the text to make it more clear what is actually happening.

yxxhero commented:
did you test it locally?

yxxhero commented on diff/diff.go:
```diff
@@ -141,7 +141,11 @@ func doSuppress(report Report, suppressedOutputLineRegex []string) (Report, erro
+				filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, "MODIFY_SUPPRESSED")
+			} else {
+				filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, entry.changeType)
+			}
```
so many `if`? what can we optimize?

TheMeier replied:
Well there are 3 cases here, how could you satisfy that with less  then 2 `if`s?

We rewrite it a bit:
```
		if !containsDiff && entry.changeType == "MODIFY" {
			entry.changeType = "MODIFY_SUPPRESSED"
		}
		if containsDiff {
			filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, diffs, entry.changeType)
		} else {
  		filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, entry.changeType)
		}
```
or
```
		if containsDiff {
			filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, diffs, entry.changeType)
		} else {
		  if entry.changeType == "MODIFY" {
				entry.changeType = "MODIFY_SUPPRESSED"
			}
  		filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, entry.changeType)
		}
```

TheMeier commented:
> did you test it locally?

I did add a unit test.

yxxhero commented:
```
		if !containsDiff && entry.changeType == "MODIFY" {
			entry.changeType = "MODIFY_SUPPRESSED"
		}
		if containsDiff {
			filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, diffs, entry.changeType)
		} else {
  		filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, entry.changeType)
		}

```

this is better.

yxxhero commented:
or we can use new var?

TheMeier commented:
I have changed the code. 
I don't see how/where a new variable would change anything.

How about this variant:
```
switch {
case containsDiff:
    filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, diffs, entry.changeType)
case entry.changeType == "MODIFY":
    // No diffs and type is MODIFY, change type to MODIFY_SUPPRESSED
    entry.changeType = "MODIFY_SUPPRESSED"
    filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, entry.changeType)
default:
    // No diffs, not MODIFY type
    filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, []difflib.DiffRecord{}, entry.changeType)
}
```

yxxhero commented:
> I have changed the code. I don't see how/where a new variable would change anything.
> 
> How about this variant:
> 
> ```
> 		var diffRecords []difflib.DiffRecord
> 		switch {
> 		case containsDiff:
> 			diffRecords = diffs
> 		case entry.changeType == "MODIFY":
> 			entry.changeType = "MODIFY_SUPPRESSED"
> 			diffRecords = []difflib.DiffRecord{}
> 		default:
> 			diffRecords = []difflib.DiffRecord{}
> 		}
> 
> 		filteredReport.addEntry(entry.key, entry.suppressedKinds, entry.kind, entry.context, diffRecords, entry.changeType)
> ```

we can try this.

TheMeier commented:
@yxxhero I have simplified the code even more using switch/case. Also I fixed the the color to be actually used and chose a color that actually exists in the lib.
unit tests are ok for me, did some manual tests which also are ok for me. So in my opinion this is now good to go

yxxhero reviewed (approved):
LGTM

yxxhero commented:
@mumoshu WDYT?
